src/analysis/normed_space/dual.lean

```diff
@@ -0,0 +1,81 @@
+/-
+Copyright (c) 2020 Heather Macbeth. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Heather Macbeth
+-/
+import analysis.normed_space.hahn_banach
+
+/-!
+# The topological dual of a normed space
+
+In this file we define the topological dual of a normed space, and the bounded linear map from
+a normed space into its double dual.
+
+We also prove that, for base field the real numbers, this map is an isometry.  (TODO:  the same for the
+complex numbers.)
+-/
+
+noncomputable theory
+
+section general
+variables (𝕜 : Type*) [nondiscrete_normed_field 𝕜]
+variables (E : Type*) [normed_group E] [normed_space 𝕜 E]
+
+/-- The topological dual of a normed space `E`. -/
+@[derive [normed_group, normed_space 𝕜]] def dual := E →L[𝕜] 𝕜
```

Comment:
What is its name currently?  Is it `normed_space.dual.dual`, derived from the filename?  If so, indeed it seems like a good idea to use the `normed_space` namespace, probably all through the file?

Reply:
Right now it's just called `dual`, no namespace. (Sometimes we'll say it's in the root namespace, since you can write `_root_.dual` to disambuguate, but this usually isn't needed.)

Namespaces aren't semantically very meaningful. Every declaration has a name, which is effectively a list of strings, e.g. `["module", "dual"]`. You could put this in the `normed_space` namespace just by writing `def normed_space.dual` instead. But I think you're right, everything in the file belongs in this namespace. So `namespace normed_space` at the top and `end normed_space` at the bottom should do the trick.

Reply:
I phrased this badly, but I meant that to refer to the things in this file from outside the file, one would need to call them as `normed_space.dual.foo`; is that correct?

Anyway, have added the `normed_space` namespace throughout, as of 07bf973.